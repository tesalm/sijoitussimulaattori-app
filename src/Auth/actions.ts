import { to } from 'await-to-js';
import firebase from 'react-native-firebase';
import { Dispatch } from 'redux';

import { t } from '../assets/i18n';
import { getCurrentUser, getUserData, signInAnonymously, signOut, upsertUserData } from '../firestore';
import { User, UserAuth } from '../models';
import { createDefaultUser } from '../util/users';
import { UserData } from './../models/user.model';

export enum ActionType {
  RestoreLoginSuccess = '[Login] Restore Login Success',
  RestoreLoginImpossible = '[Login] Restore Login Impossible',
  LoginRequest = '[Login] Login Request',
  LoginSuccess = '[Login] Login Success',
  LoginFailure = '[Login] Login Failure',
  LogoutRequest = '[Login] Logout Request',
  FetchUserDataSuccess = '[Login] Fetch User Data Success',
  FetchUserDataFailure = '[Login] Fetch User Data Failure',
  DeleteCurrentUserRequest = '[Login] Delete Current User Request',
  DeleteCurrentUserSuccess = '[Login] Delete Current User Success',
  DeleteCurrentUserFailure = '[Login] Delete Current User Failure',
}

export type AuthAction =
  | RestoreLoginSuccess
  | RestoreLoginImpossible
  | LoginRequest
  | LoginSuccess
  | LoginFailure
  | LogoutRequest
  | FetchUserDataSuccess
  | FetchUserDataFailure
  | DeleteCurrentUserRequest
  | DeleteCurrentUserSuccess
  | DeleteCurrentUserFailure;

export class LoginRequest {
  readonly type = ActionType.LoginRequest;
  constructor() {
    return { type: this.type };
  }
}

export class LoginSuccess {
  readonly type = ActionType.LoginSuccess;
  constructor(public user: User) {
    return { type: this.type, user };
  }
}

export class LoginFailure {
  readonly type = ActionType.LoginFailure;
  constructor(public error: Error) {
    return { type: this.type, error };
  }
}

export class LogoutRequest {
  readonly type = ActionType.LogoutRequest;
  constructor() {
    return { type: this.type };
  }
}

export class FetchUserDataSuccess {
  readonly type = ActionType.FetchUserDataSuccess;
  constructor(public user: User) {
    return { type: this.type, user };
  }
}

export class FetchUserDataFailure {
  readonly type = ActionType.FetchUserDataFailure;
  constructor(public userDataError: Error) {
    return { type: this.type, userDataError };
  }
}

export class RestoreLoginSuccess {
  readonly type = ActionType.RestoreLoginSuccess;
  constructor(public userAuth: UserAuth) {
    return { type: this.type, userAuth };
  }
}

export class RestoreLoginImpossible {
  readonly type = ActionType.RestoreLoginImpossible;
  constructor() {
    return { type: this.type };
  }
}

export class DeleteCurrentUserRequest {
  readonly type = ActionType.DeleteCurrentUserRequest;
  constructor() {
    return { type: this.type };
  }
}

export class DeleteCurrentUserSuccess {
  readonly type = ActionType.DeleteCurrentUserSuccess;
  constructor() {
    return { type: this.type };
  }
}

export class DeleteCurrentUserFailure {
  readonly type = ActionType.DeleteCurrentUserFailure;
  constructor(public error: Error) {
    return { type: this.type, error: this.error };
  }
}
const restorePreviousLogin = () => async (dispatch: Dispatch<AuthAction>) => {
  // Wait the completion of the Firebase initialization and catch the logged in user
  // when after it.
  const unsubscripe = firebase
    .auth()
    .onAuthStateChanged((userAuth: UserAuth) => {
      if (userAuth) {
        // Previous user can be restored.
        dispatch(new RestoreLoginSuccess(userAuth));
        fetchUserData(dispatch, userAuth);
      } else {
        // There is no previous user to restore.
        dispatch(new RestoreLoginImpossible());
      }

      // Catch only the first authentication state change event.
      unsubscripe();
    });
};

// This function should be used only for "old" users.
const fetchUserData = async (
  dispatch: Dispatch<AuthAction>,
  userAuth: UserAuth
) => {
  let userData: UserData = {};

  try {
    userData = (await getUserData(userAuth.uid)) || userData;
  } catch (err) {
    dispatch(new FetchUserDataFailure(err));
    return; // <- Exit on error.
  }

  const user: User = {
    ...userAuth,
    ...userData,
  };

  dispatch(new FetchUserDataSuccess(user));
};

const login = () => async (dispatch: Dispatch<AuthAction>) => {
  dispatch(new LoginRequest());

  const [err, fsUser] = await to(signInAnonymously());

  if (err || !fsUser) {
    return dispatch(new LoginFailure(err || Error('User not found.')));
  }

  const userAuth: UserAuth = {
    uid: fsUser.user.uid,
  };

  // It should be now that the login was successfull.
  // The rest of this function will handle user data.
  dispatch(new LoginSuccess(userAuth));

  let userData: UserData = {};

  if (fsUser.additionalUserInfo && fsUser.additionalUserInfo.isNewUser) {
    // Create new user.
    userData = createDefaultUser();

    try {
      await upsertUserData(userAuth.uid, userData);
    } catch (err) {
      console.error('Error creating defaut user:' + err); // TODO: Error handling needed? Maybe dispatch failure notification action
    }
  } else {
    // Fetch data for "old" user.
    fetchUserData(dispatch, userAuth);
    return; // <- !
  }

  const user: User = {
    ...userAuth,
    ...userData,
  };

  // Assign user data for new user. The user might got nothing as UserData, if something
  // went wrong on the user creation initialization.
  dispatch(new FetchUserDataSuccess(user));
};

const logout = () => async (dispatch: Dispatch<LogoutRequest>) => {
  const [err] = await to(signOut());

  if (err) {
    console.error('Error logging out');
  } else {
    dispatch(new LogoutRequest());
  }
};

const deleteCurrentUser = () => async (dispatch: Dispatch<AuthAction>) => {
  dispatch(new DeleteCurrentUserRequest());

  const currentUser = getCurrentUser();

  if (!currentUser) {
    dispatch(
      new DeleteCurrentUserFailure(new Error(t('Auth.NoAuthenticatedUser')))
    );
    return;
  }

  const [err] = await to(currentUser.delete());

  if (err) {
    dispatch(new DeleteCurrentUserFailure(err));
  } else {
    dispatch(new DeleteCurrentUserSuccess());
  }
};

export { login, logout, deleteCurrentUser, restorePreviousLogin };
