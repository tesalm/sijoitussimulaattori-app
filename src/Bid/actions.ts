import { Dispatch } from 'redux';
import { BidInfo } from './reducer';

export enum ActionType {
  UpdateAction = '[Bid] Update action',
  UpdateBidLevel = '[Bid] Update bid level',
  UpdateSumOfStocks = '[Bid] Update sum of stocks',
  UpdateSelectedPortfolio = '[Bid] Update selected portfolio',
  SetSymbol = '[Bid] Set symbol',
}

export type BidAction =
  | UpdateAction
  | UpdateBidLevel
  | UpdateSumOfStocks
  | UpdateSelectedPortfolio
  | SetSymbol;

export class UpdateAction {
  readonly type = ActionType.UpdateAction;
  constructor(public actionType: string) {
    return { type: this.type, actionType };
  }
}

export class UpdateBidLevel {
  readonly type = ActionType.UpdateBidLevel;
  constructor(public bidLevel: number) {
    return { type: this.type, bidLevel };
  }
}

export class UpdateSumOfStocks {
  readonly type = ActionType.UpdateSumOfStocks;
  constructor(public sumOfStocks: number) {
    return { type: this.type, sumOfStocks };
  }
}

export class UpdateSelectedPortfolio {
  readonly type = ActionType.UpdateSelectedPortfolio;
  constructor(public selectedPortfolio: string) {
    return { type: this.type, selectedPortfolio };
  }
}

export class SetSymbol {
  readonly type = ActionType.SetSymbol;
  constructor(public symbol: string) {
    return { type: this.type, symbol };
  }
}

const updateAction = (action: string) => async (
  dispatch: Dispatch<BidAction>
) => {
  dispatch(new UpdateAction(action));
};

const updateBidLevel = (bidLevel: number) => async (
  dispatch: Dispatch<BidAction>
) => {
  dispatch(new UpdateBidLevel(bidLevel));
};

const updateSumOfStocks = (sumOfStocks: number) => async (
  dispatch: Dispatch<BidAction>
) => {
  dispatch(new UpdateSumOfStocks(sumOfStocks));
};

const updateSelectedPortfolio = (selectedPortfolio: string) => async (
  dispatch: Dispatch<BidAction>
) => {
  dispatch(new UpdateSelectedPortfolio(selectedPortfolio));
};

const saveBidForm = (
  action: string,
  bidLevel: number,
  sumOfStocks: number,
  selectedPortfolio: string,
  symbol: string
) => async (dispatch: Dispatch<BidAction>) => {
  dispatch(new UpdateAction(action));
  dispatch(new UpdateBidLevel(bidLevel));
  dispatch(new UpdateSumOfStocks(sumOfStocks));
  dispatch(new UpdateSelectedPortfolio(selectedPortfolio));
  dispatch(new SetSymbol(symbol));
};

export {
  updateAction,
  updateBidLevel,
  updateSumOfStocks,
  updateSelectedPortfolio,
  saveBidForm,
};
