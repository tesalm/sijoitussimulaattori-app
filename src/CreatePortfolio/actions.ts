import { Dispatch } from 'redux';
import { createPortfolioApiRequest } from '../utils/api';

export enum ActionType {
  CreatePortfolioBegin = '[CreatingPortfolio] Creating portfolio begin',
  CreatePortfolioSuccess = '[CreatingPortfolio] Creating portfolio success',
  CreatePortfolioFailure = '[CreatingPortfolio] Creating portfolio failure',
}

export type CreatePortfolioAction =
  | CreatePortfolioBegin
  | CreatePortfolioSuccess
  | CreatePortfolioFailure;

export class CreatePortfolioBegin {
  readonly type = ActionType.CreatePortfolioBegin;
  constructor() {
    return { type: this.type };
  }
}

export class CreatePortfolioSuccess {
  readonly type = ActionType.CreatePortfolioSuccess;
  constructor() {
    return { type: this.type };
  }
}

export class CreatePortfolioFailure {
  readonly type = ActionType.CreatePortfolioFailure;
  constructor(public error: Error) {
    return { type: this.type, error };
  }
}

const sendPortfolioInfo = (name: string, amount: number) => async (
  dispatch: Dispatch<CreatePortfolioAction>
) => {
  dispatch(new CreatePortfolioBegin());
  try {
    await createPortfolioApiRequest(name, amount);
    dispatch(new CreatePortfolioSuccess());
  } catch (error) {
    dispatch(new CreatePortfolioFailure(error));
  }
};

export { sendPortfolioInfo };
