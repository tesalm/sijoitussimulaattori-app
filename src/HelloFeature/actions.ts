import { Dispatch } from 'redux';

export enum ActionType {
  AddToCounter = 'ADD_TO_COUNTER',
  RemoveFromCounter = 'REMOVE_FROM_COUNTER',
}

export type CounterAction = AddToCounter | RemoveFromCounter;

interface AddToCounter {
  type: ActionType.AddToCounter;
  amount: number;
}

interface RemoveFromCounter {
  type: ActionType.RemoveFromCounter;
}

const addToCounterAction = (amount: number): AddToCounter => ({
  type: ActionType.AddToCounter,
  amount: amount,
});

const removeFromCounterAction = (): RemoveFromCounter => ({
  type: ActionType.RemoveFromCounter,
});

const addToCounter = (amount: number) => async (
  dispatch: Dispatch<AddToCounter>
) => dispatch(addToCounterAction(amount));

const removeFromCounter = () => (dispatch: Dispatch<RemoveFromCounter>) =>
  dispatch(removeFromCounterAction());

export { addToCounter, removeFromCounter };
