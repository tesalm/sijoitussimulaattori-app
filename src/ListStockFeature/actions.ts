import { Dispatch } from 'redux';
import Config from 'react-native-config'


export enum ActionType {
    GetStocksBegin = 'GET_STOCKS_BEGIN',
    GetStocksSuccess = 'GET_STOCKS_SUCCESS',
    GetStocksFailure = 'GET_STOCKS_FAILURE',
  }

export type StocksAction = GetStocksBegin | GetStocksSuccess | GetStocksFailure;

export class GetStocksBegin {
    readonly type = ActionType.GetStocksBegin;
    constructor(){
        return {type: this.type}
    }
}

export class GetStocksSuccess {
    readonly type = ActionType.GetStocksSuccess;
    constructor(public stocks :Array<{key: string, revenue: number, lastsale: number}> ){
        return {type: this.type, stocks}
    }
}

export class GetStocksFailure {
    readonly type = ActionType.GetStocksFailure;
    constructor(public error: Error){
        return {type: this.type, error}
    }
}
//This is called by StockListing. Dispatch needed actions and fetch the data
const getStocks = ()  => async(dispatch:Dispatch<StocksAction>) => {
    dispatch(new GetStocksBegin());
    fetch(Config.STOCK_API_URL).then(res => res.json())
    .then(json => {dispatch(new GetStocksSuccess(json.results))})
    .catch(error => {dispatch(new GetStocksFailure(error))})

}

export{getStocks}