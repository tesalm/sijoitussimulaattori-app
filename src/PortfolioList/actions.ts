import { Dispatch } from 'redux';

import { portfolioApiRequest, portfolioListApiRequest } from '../utils/api';
import { Portfolio, SinglePortfolio } from './reducers';

export enum ActionType {
  RequestPortfolioListingBegin = '[Portfolios] API Request',
  RequestPortfoliosListingSuccess = '[Portfolios] API Success',
  RequestPortfoliosListingFailure = '[Portfolios] API Failure',
  RequestPortfolioBegin = '[Portfolio] Data Request',
  RequestPortfolioSuccess = '[Portfolio] Data Success',
  RequestPortfolioFailure = '[Portfolio] Data Failure',
  SaveId = '[Portfolios] Save Active Portfolio Id',
}

export type PortfolioAction =
  | RequestPortfolioListingBegin
  | RequestPortfoliosListingSuccess
  | RequestPortfoliosListingFailure
  | SaveId
  | RequestPortfolioBegin
  | RequestPortfolioSuccess
  | RequestPortfolioFailure;

export class RequestPortfolioListingBegin {
  readonly type = ActionType.RequestPortfolioListingBegin;
  constructor() {
    return { type: this.type };
  }
}

export class RequestPortfoliosListingSuccess {
  readonly type = ActionType.RequestPortfoliosListingSuccess;
  constructor(public portfolios: Array<SinglePortfolio>) {
    return { type: this.type, portfolios };
  }
}

export class RequestPortfoliosListingFailure {
  readonly type = ActionType.RequestPortfoliosListingFailure;
  constructor(public error: Error) {
    return { type: this.type, error };
  }
}

export class SaveId {
  readonly type = ActionType.SaveId;
  constructor(public id: string) {
    return { type: this.type, id };
  }
}

export class RequestPortfolioBegin {
  readonly type = ActionType.RequestPortfolioBegin;
  constructor(public portfolioId: string) {
    return { type: this.type, portfolioId };
  }
}

export class RequestPortfolioSuccess {
  readonly type = ActionType.RequestPortfolioSuccess;
  constructor(public portfolio: Portfolio, public portfolioId: string) {
    return { type: this.type, portfolio, portfolioId };
  }
}

export class RequestPortfolioFailure {
  readonly type = ActionType.RequestPortfolioFailure;
  constructor(public error: Error, public portfolioId: string) {
    return { type: this.type, error, portfolioId };
  }
}

const getPortfolioData = (portfolioId: string) => async (
  dispatch: Dispatch<PortfolioAction>
) => {
  dispatch(new RequestPortfolioBegin(portfolioId));
  try {
    const data = await portfolioApiRequest(portfolioId);
    dispatch(new RequestPortfolioSuccess(data, portfolioId));
  } catch (error) {
    dispatch(new RequestPortfolioFailure(error, portfolioId));
  }
};

const SaveAsCurrentPortfolioId = (symbol: string) => async (
  dispatch: Dispatch<PortfolioAction>
) => {
  dispatch(new SaveId(symbol));
};

const getPortfolios = () => async (dispatch: Dispatch<PortfolioAction>) => {
  dispatch(new RequestPortfolioListingBegin());
  try {
    const data = await portfolioListApiRequest();
    data.forEach((portfolio) => {
      portfolio.portfolioInfo = {
        loading: false,
        error: undefined,
        refreshing: false,
        portfolio: undefined,
      };
    });
    dispatch(new RequestPortfoliosListingSuccess(data));
  } catch (error) {
    dispatch(new RequestPortfoliosListingFailure(error));
  }
};

export { getPortfolios, SaveAsCurrentPortfolioId, getPortfolioData };
