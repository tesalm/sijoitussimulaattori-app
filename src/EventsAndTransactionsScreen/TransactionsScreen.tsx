import React from 'react';
import { RefreshControl, ScrollView, Text, View } from 'react-native';
import { Card } from 'react-native-elements';
import { NavigationScreenProps } from 'react-navigation';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';

import { Colors } from '../App/colors';
import { cardStyles } from '../App/styles';
import { getTransactions } from '../PortfolioList/actions';
import { SinglePortfolio } from '../PortfolioList/reducer';
import { RootState } from '../redux/reducers';
import { t } from './../assets/i18n';
import { transactionStyles } from './styles';

export interface TransactionsProps {
  portfolio?: SinglePortfolio;
  portfolioId?: string;
  getTransactions: typeof getTransactions;
}

type TransactionsPropsWithNavigation = TransactionsProps &
  NavigationScreenProps;

export class TransactionsScreen extends React.Component<
  TransactionsPropsWithNavigation
> {
  static navigationOptions = { title: t('PortfolioPage.Events') };
  constructor(props: TransactionsPropsWithNavigation) {
    super(props);
  }

  componentDidMount() {
    const { portfolio, portfolioId } = this.props;
    if (
      portfolio &&
      portfolioId &&
      portfolio.transactions.transactionListing.length < 1
    ) {
      this.props.getTransactions(portfolioId);
    }
  }

  refreshTransactions = () => {
    if (this.props.portfolioId) {
      this.props.getTransactions(this.props.portfolioId);
    }
  };

  renderContent = () => {
    const { portfolio, portfolioId } = this.props;
    if (portfolio === undefined || portfolioId === undefined) {
      return <Text>Error, portfolio not found! </Text>;
    }
    const { transactionListing } = portfolio.transactions;
    if (portfolio.transactions.error) {
      return (
        <Text style={transactionStyles.noActionsText}>
          Error! {portfolio.transactions.error.message}
        </Text>
      );
    }
    if (transactionListing.length === 0) {
      return (
        <Text style={transactionStyles.noActionsText}>
          {t('TransactionsPage.NoTransactions')}
        </Text>
      );
    }

    return transactionListing.map((item, i) => (
      <View key={i}>
        {item.status !== 'MARKET' && (
          <View style={transactionStyles.eventContainer}>
            <View style={transactionStyles.eventSection}>
              <View style={{ flex: 1 }}>
                {item.status === 'CANCELLED' ? (
                  <Text style={transactionStyles.bold}>
                    {new Date(item.cancelledAt).getDay()}
                    {'.'}
                    {new Date(item.cancelledAt).getMonth()}
                    {'.'}
                    {new Date(item.cancelledAt).getFullYear()}
                  </Text>
                ) : (
                  <Text style={transactionStyles.bold}>
                    {new Date(item.fulfilledAt).getDay()}
                    {'.'}
                    {new Date(item.fulfilledAt).getMonth()}
                    {'.'}
                    {new Date(item.fulfilledAt).getFullYear()}
                  </Text>
                )}
              </View>

              <View style={{ flex: 1 }}>
                <Text style={transactionStyles.bold}>{item.type}</Text>
              </View>
              <View style={{ flex: 1 }}>
                <Text style={transactionStyles.bold}>{item.symbol}</Text>
              </View>
            </View>
            <View style={transactionStyles.eventSection}>
              <View style={{ flex: 1 }}>
                <Text style={transactionStyles.basicText}>
                  {t('TransactionsPage.Amount')}
                </Text>
                <Text style={transactionStyles.bold}>{item.amount}</Text>
              </View>
              <View style={{ flex: 1 }}>
                <Text style={transactionStyles.basicText}>
                  {t('TransactionsPage.Value')}
                </Text>
                <Text style={transactionStyles.bold}>{item.price}</Text>
              </View>
              <View style={{ flex: 1 }}>
                <Text style={transactionStyles.basicText}>
                  {t('TransactionsPage.State')}
                </Text>
                <Text style={transactionStyles.bold}>{item.status}</Text>
              </View>
            </View>
          </View>
        )}
      </View>
    ));
  };

  render() {
    const { portfolio } = this.props;
    if (portfolio) {
      return (
        <ScrollView
          refreshControl={
            <RefreshControl
              refreshing={portfolio.transactions.loading}
              onRefresh={this.refreshTransactions}
              colors={[Colors.baseColor]}
            />
          }
        >
          <Card containerStyle={cardStyles.container}>
            {this.renderContent()}
          </Card>
        </ScrollView>
      );
    } else {
      // TODO: Format the error message to user
      return <Text>Error, portfolio not found! </Text>;
    }
  }
}

const mapStateToProps = (state: RootState) => ({
  portfolioId: state.portfolioListing.portfolioId,
  portfolio: state.portfolioListing.portfolioListing.find((portfolio) => {
    return portfolio.uid === state.portfolioListing.portfolioId;
  }),
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      getTransactions: getTransactions,
    },
    dispatch
  );
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TransactionsScreen);
