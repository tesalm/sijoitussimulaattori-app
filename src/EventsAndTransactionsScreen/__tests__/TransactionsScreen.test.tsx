import { shallow } from 'enzyme';
import * as React from 'react';
import configureStore from 'redux-mock-store';

import { getTransactions } from '../../PortfolioList/actions';
import { TransactionsProps, TransactionsScreen } from '../TransactionsScreen';

import 'react-native';

describe('Transactions screen tests: ', () => {
  const navigationMock: any = {};
  const middlewares: any = [];
  const mockStore = configureStore(middlewares);

  const defaultTransactionProps: TransactionsProps = {
    portfolio: {
      uid: 'Portfolio1',
      name: 'Portfolio1',
      balance: 500,
      totalRevenue: -0.02,
      totalMarketValue: 5000,
      lastDayRevenue: 0,
      ownerId: 'JohnDoe',
      portfolioInfo: {
        loading: false,
        error: undefined,
        refreshing: false,
        portfolio: undefined,
      },
      transactionInfo: {
        transactions: undefined,
        transactionsLoading: false,
        transactionsError: undefined,
        transactionSuccess: false,
      },
      transactions: {
        transactionListing: [],
        loading: false,
        error: undefined,
      },
    },
    portfolioId: 'Portfolio1',
    getTransactions: getTransactions,
  };

  const successTransactionProps: TransactionsProps = {
    portfolio: {
      uid: 'Portfolio1',
      name: 'Portfolio1',
      balance: 500,
      totalRevenue: -0.02,
      totalMarketValue: 5000,
      lastDayRevenue: 0,
      ownerId: 'JohnDoe',
      portfolioInfo: {
        loading: false,
        error: undefined,
        refreshing: false,
        portfolio: {
          uid: 'Portfolio1',
          name: 'Portfolio1',
          balance: 500,
          totalRevenue: -0.02,
          totalMarketValue: 5000,
          lastDayRevenue: 0,
          ownerId: 'JohnDoe',
          stocks: [],
        },
      },
      transactionInfo: {
        transactions: undefined,
        transactionsLoading: false,
        transactionsError: undefined,
        transactionSuccess: false,
      },
      transactions: {
        transactionListing: [
          {
            uid: '1',
            type: 'BUY',
            symbol: 'AMZN',
            amount: 20,
            price: 222.2,
            expiresAt: '2018-12-14T15:55:00.000Z',
            status: 'CANCELLED',
            fulfilledAt: '',
            cancelledAt: '2018-12-14T15:55:00.000Z',
          },
          {
            uid: '2',
            type: 'BUY',
            symbol: 'GOOGL',
            amount: 30,
            price: 333.3,
            expiresAt: '2018-12-14T15:55:00.000Z',
            status: 'FULFILLED',
            fulfilledAt: '2018-12-14T15:55:00.000Z',
            cancelledAt: '',
          },
        ],
        loading: false,
        error: undefined,
      },
    },
    portfolioId: 'Portfolio1',
    getTransactions: getTransactions,
  };

  const errorTransactionProps: TransactionsProps = {
    portfolio: {
      uid: 'Portfolio1',
      name: 'Portfolio1',
      balance: 500,
      totalRevenue: -0.02,
      totalMarketValue: 5000,
      lastDayRevenue: 0,
      ownerId: 'JohnDoe',
      portfolioInfo: {
        loading: false,
        error: undefined,
        refreshing: false,
        portfolio: undefined,
      },
      transactionInfo: {
        transactions: undefined,
        transactionsLoading: false,
        transactionsError: undefined,
        transactionSuccess: false,
      },
      transactions: {
        transactionListing: [],
        loading: false,
        error: {
          name: 'Network Error',
          message: 'Network connection failed',
        },
      },
    },
    portfolioId: 'Portfolio1',
    getTransactions: jest.fn(),
  };

  it('renders correctly', async () => {
    const component = shallow(
      <TransactionsScreen {...defaultTransactionProps} {...navigationMock} />,
      { context: { store: mockStore() } }
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with error', async () => {
    const component = shallow(
      <TransactionsScreen {...errorTransactionProps} {...navigationMock} />,
      { context: { store: mockStore() } }
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with data', async () => {
    const component = shallow(
      <TransactionsScreen {...successTransactionProps} {...navigationMock} />,
      { context: { store: mockStore() } }
    );
    expect(component).toMatchSnapshot();
  });
});
