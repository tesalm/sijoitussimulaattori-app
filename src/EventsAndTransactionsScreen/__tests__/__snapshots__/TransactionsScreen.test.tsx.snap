// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions screen tests:  renders correctly 1`] = `
<ScrollView
  refreshControl={
    <RefreshControlMock
      colors={
        Array [
          "#004D40",
        ]
      }
      onRefresh={[Function]}
      refreshing={false}
    />
  }
>
  <Themed.Card
    containerStyle={
      Object {
        "flex": 1,
        "margin": 0,
        "paddingBottom": 32,
        "paddingLeft": 32,
        "paddingRight": 32,
        "paddingTop": 32,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Roboto",
          "fontSize": 15,
          "marginBottom": 23.541176470588233,
          "marginTop": 23.541176470588233,
          "opacity": 0.6,
        }
      }
    >
      No transactions
    </Text>
  </Themed.Card>
</ScrollView>
`;

exports[`Transactions screen tests:  renders correctly with data 1`] = `
<ScrollView
  refreshControl={
    <RefreshControlMock
      colors={
        Array [
          "#004D40",
        ]
      }
      onRefresh={[Function]}
      refreshing={false}
    />
  }
>
  <Themed.Card
    containerStyle={
      Object {
        "flex": 1,
        "margin": 0,
        "paddingBottom": 32,
        "paddingLeft": 32,
        "paddingRight": 32,
        "paddingTop": 32,
      }
    }
  >
    <View
      key="0"
    >
      <View
        style={
          Object {
            "marginBottom": 23.541176470588233,
            "marginTop": 23.541176470588233,
          }
        }
      >
        <View
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 11.770588235294117,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              5
              .
              11
              .
              2018
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              BUY
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              AMZN
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 11.770588235294117,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 12,
                  "textAlign": "justify",
                }
              }
            >
              Amount
            </Text>
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              20
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 12,
                  "textAlign": "justify",
                }
              }
            >
              Value
            </Text>
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              222.2
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 12,
                  "textAlign": "justify",
                }
              }
            >
              State
            </Text>
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              CANCELLED
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      key="1"
    >
      <View
        style={
          Object {
            "marginBottom": 23.541176470588233,
            "marginTop": 23.541176470588233,
          }
        }
      >
        <View
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 11.770588235294117,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              5
              .
              11
              .
              2018
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              BUY
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              GOOGL
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 11.770588235294117,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 12,
                  "textAlign": "justify",
                }
              }
            >
              Amount
            </Text>
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              30
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 12,
                  "textAlign": "justify",
                }
              }
            >
              Value
            </Text>
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              333.3
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 12,
                  "textAlign": "justify",
                }
              }
            >
              State
            </Text>
            <Text
              style={
                Object {
                  "color": "#004D40",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "justify",
                }
              }
            >
              FULFILLED
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Themed.Card>
</ScrollView>
`;

exports[`Transactions screen tests:  renders correctly with error 1`] = `
<ScrollView
  refreshControl={
    <RefreshControlMock
      colors={
        Array [
          "#004D40",
        ]
      }
      onRefresh={[Function]}
      refreshing={false}
    />
  }
>
  <Themed.Card
    containerStyle={
      Object {
        "flex": 1,
        "margin": 0,
        "paddingBottom": 32,
        "paddingLeft": 32,
        "paddingRight": 32,
        "paddingTop": 32,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Roboto",
          "fontSize": 15,
          "marginBottom": 23.541176470588233,
          "marginTop": 23.541176470588233,
          "opacity": 0.6,
        }
      }
    >
      Error! 
      Network connection failed
    </Text>
  </Themed.Card>
</ScrollView>
`;
